src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -128,20 +102,7 @@ class SidebarLinks extends React.Component {
         }
 
         Timing.start(CONST.TIMING.SIDEBAR_LINKS_FILTER_REPORTS);
-        const optionListItems = this.getRecentReportsOptionListItems(
-            this.props.currentlyViewedReportID,
-            this.props.priorityMode,
-            this.props.reports,
-            this.props.personalDetails,
-            this.props.betas,
-            this.props.reportActions,
-        );
-        const sections = [{
-            title: '',
-            indexOffset: 0,
-            data: optionListItems,
-            shouldShow: true,
-        }];
+        const optionListItems = OptionsListUtilsLHN.getOrderedReportIDs();
```

Comment:
Since this code depends on having two subscriptions (the `withOnyx` call in `SidebarLinks` and the `Onyx.connect()` calls in `SidebarUtils`) it might be helpful to add some kind of comment to explain how this all works.

If I was seeing this for the first time I wouldn't necessarily make the connection that this is intended to run whenever the component's props change and that the options will be updated because the `SidebarUtils` is subscribing to the same data.

Separately, when reviewing the current subscriptions here I see that we are missing a few things:

- ONYXKEYS.COLLECTION.POLICY
- ONYXKEYS.COLLECTION.REPORT_IOUS
- ONYXKEYS.SESSION
- ONYXKEYS.NVP_PREFERRED_LOCALE

Do we need to add those to the `withOnyx()` mapping below?

Reply:
I can try to add some comments to clarify what's happening, sure.

As for those subscriptions

- ONYXKEYS.COLLECTION.POLICY - The only thing that becomes stale here is the policy name and icon, so it probably should be added.
- ONYXKEYS.COLLECTION.REPORT_IOUS - I _think_ that any changes to an IOU would also trigger a change to the report collection, so not strictly needed and would maybe cause more re-renders than necessary
- ONYXKEYS.SESSION - The only thing that becomes stale is `currentUserLogin` and if that changed, then they would have done a full sign-out, so subscribing to it isn't necessary
- ONYXKEYS.NVP_PREFERRED_LOCALE - The archive reason becomes stale here, so this also needs added

Good catch on those!

Reply:
>I think that any changes to an IOU would also trigger a change to the report collection, so not strictly needed and would maybe cause more re-renders than necessary

Ah hmm that's not super obvious and feels kind of brittle if someone decides that just the iou report _should_ update without updating the associated report. Also sort of leads me to another question about whether updating a few of these Onyx keys at once is a general performance bottleneck or not. And whether we should address that as a more general problem vs. trying to optimize by timing the updates of different keys to happen together.

Reply:
> * ONYXKEYS.COLLECTION.POLICY - The only thing that becomes stale here is the policy name and icon, so it probably should be added.

Yes, please add that. It will fix this issue https://github.com/Expensify/App/issues/10725